apiserver/pkg/storage/calico/resource.go

```diff
@@ -188,7 +189,7 @@ func checkPreconditions(key string, preconditions *storage.Preconditions, out ru
 	if err != nil {
 		return storage.NewInternalErrorf("can't enforce preconditions %v on un-introspectable object %v, got error: %v", *preconditions, out, err)
 	}
-	if preconditions.UID != nil && *preconditions.UID != objMeta.GetUID() {
+	if preconditions.UID != nil && *preconditions.UID != resources.ReverseUID(objMeta.GetUID()) {
```

Comment:
I don't think this code is hit when calico-node updates custom resources; I think it goes through libcalico-go directly?